src/utils/response/responseHandlers.ts

```diff
@@ -47,41 +64,79 @@ export async function toApiResponse<T>(promise: RequestPromise<T>): Promise<ApiR
 }
 
 /**
- * @return A promise that resolves to T or rejects if no results
+ * @return A promise that resolves to Type or rejects if no results
  */
-export async function required<T>(promise: RequestPromise<T[]>): Promise<T> {
-  const response = await toApiResponse<T[]>(promise);
+export async function required<ResourceName extends ResourceNames>(
+  promise: RequestPromise<ResourceName>
+): Required<ResourceName, Resources[ResourceName][]> {
+  const response = await toApiResponse<ResourceName>(promise);
 
   if (response.NumResults > 0) {
-    return response.Results[0];
+    return {
+      Results: response.Results,
+      Affected: response.Affected ?? {},
+      Deleted: response.Deleted ?? {},
+    };
   } else {
     throw new Error('There are no results.');
   }
 }
 
-export async function requiredSingle<T>(promise: RequestPromise<T>): Promise<T> {
-  const response = await toApiResponse<T>(promise);
+export async function requiredSingle<ResourceName extends ResourceNames>(
+  promise: RequestPromise<ResourceName>
+): Promise<LibraryReturn<ResourceName>> {
+  const response = await toApiResponse<ResourceName>(promise);
 
   if (response.NumResults > 0) {
-    return response.Results;
+    return {
+      Results: response.Results[0],
+      Affected: response.Affected ?? {},
+      Deleted: response.Deleted ?? {},
+    };
   } else {
     throw new Error('There are no results.');
   }
 }
+
+export async function serverCommunication<ResourceName extends ResourceNames>(
+  promise: RequestPromise<ResourceName>
+): Promise<LibraryReturn<ResourceName, ServerCommunication>> {
+  const response = (await toApiResponse<ResourceName>(promise)) as unknown as ApiResponseOnSuccess<ResourceName, ServerCommunication>;
+
+  if (response.Results.host) {
+    return {
+      Results: response.Results,
+      Affected: response.Affected ?? {},
```

Comment:
Should serverCommunication return also this one as we know it can not.

Reply:
No, will fix with next commit